hazelcast/src/main/java/com/hazelcast/cardinality/CardinalityEstimator.java

```diff
@@ -0,0 +1,127 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cardinality;
+
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.spi.annotation.Beta;
+
+/**
+ * CardinalityEstimator is a redundant and highly available distributed data-structure used
+ * for probabilistic cardinality estimation purposes, on unique items, in significantly sized data cultures.
+ *
+ * CardinalityEstimator is internally based on a HyperLogLog++ data-structure,
+ * and uses P^2 byte registers for storage and computation. (Default P = 14)
+ */
+@Beta
+public interface CardinalityEstimator extends DistributedObject {
+
+    /**
+     * Add a new object in the estimation set. This is the method you want to
+     * use to feed objects into the estimator.
+     *
+     * Objects are considered identical if they are serialized into the same binary blob.
+     * In other words: It does <strong>not</strong> use Java equality.
+     *
+     * @param obj object to add in the estimation set.
+     * @throws NullPointerException if obj is null
+     * @since 3.8
+     */
+    void add(Object obj);
+
+
+    /**
+     * Estimates the cardinality of the aggregation so far.
+     * If it was previously estimated and never invalidated, then a cached version is used.
+     *
+     * @return a cached estimation or a newly computed one.
+     * @since 3.8
+     */
+    long estimate();
+
+    /**
+     * Add a new object in the estimation set. This is the method you want to
+     * use to feed objects into the estimator.
+     *
+     * Objects are considered identical if they are serialized into the same binary blob.
+     * In other words: It does <strong>not</strong> use Java equality.
+     *
+     * This method will dispatch a request and return immediately an {@link ICompletableFuture}.
+     * The operations result can be obtained in a blocking way, or a
+     * callback can be provided for execution upon completion, as demonstrated in the following examples:
+     * <p>
+     * <pre>
+     *     ICompletableFuture&lt;Void&gt; future = estimator.addAsync();
+     *     // do something else, then read the result
+     *     Boolean result = future.get(); // this method will block until the result is available
+     * </pre>
+     * </p>
+     * <p>
+     * <pre>
+     *     ICompletableFuture&lt;Void&gt; future = estimator.addAsync();
+     *     future.andThen(new ExecutionCallback&lt;Void&gt;() {
+     *          void onResponse(Void response) {
+     *              // do something with the result
```

Comment:
is there any result at all? :) it's more like "at this point you know the object was added to the set"